tests/foreman/endtoend/test_cli_endtoend.py

```diff
@@ -54,339 +54,327 @@
-        """
-        result = User.info({'login': 'admin'})
-        self.assertEqual(result['login'], 'admin')
-        self.assertEqual(result['admin'], 'yes')
-
-    @skip_if_not_set('compute_resources')
-    @pytest.mark.tier4
-    @pytest.mark.upgrade
-    @pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
-    def test_positive_end_to_end(self):
-        """Perform end to end smoke tests using RH and custom repos.
-
-        1. Create a new user with admin permissions
-        2. Using the new user from above
-            1. Create a new organization
-            2. Clone and upload manifest
-            3. Create a new lifecycle environment
-            4. Create a custom product
-            5. Create a custom YUM repository
-            6. Create a custom PUPPET repository
-            7. Enable a Red Hat repository
-            8. Synchronize the three repositories
-            9. Create a new content view
-            10. Associate the YUM and Red Hat repositories to new content view
-            11. Add a PUPPET module to new content view
-            12. Publish content view
-            13. Promote content view to the lifecycle environment
-            14. Create a new activation key
-            15. Add the products to the activation key
-            16. Create a new libvirt compute resource
-            17. Create a new subnet
-            18. Create a new domain
-            19. Create a new hostgroup and associate previous entities to it
-            20. Provision a client
-
-        :id: 8c8b3ffa-0d54-436b-8eeb-1a3542e100a8
-
-        :expectedresults: All tests should succeed and Content should be
-            successfully fetched by client.
-        """
-        # step 1: Create a new user with admin permissions
-        password = gen_alphanumeric()
-        user = make_user({'admin': 'true', 'password': password})
-        user['password'] = password
-
-        # step 2.1: Create a new organization
-        org = self._create(user, Org, {'name': gen_alphanumeric()})
-
-        # step 2.2: Clone and upload manifest
-        if self.fake_manifest_is_set:
-            with manifests.clone() as manifest:
-                ssh.upload_file(manifest.content, manifest.filename)
-            Subscription.upload({'file': manifest.filename, 'organization-id': org['id']})
-
-        # step 2.3: Create a new lifecycle environment
-        lifecycle_environment = self._create(
-            user,
-            LifecycleEnvironment,
-            {'name': gen_alphanumeric(), 'organization-id': org['id'], 'prior': 'Library'},
-        )
 
-        # step 2.4: Create a custom product
-        product = self._create(
-            user, Product, {'name': gen_alphanumeric(), 'organization-id': org['id']}
-        )
-        repositories = []
 
-        # step 2.5: Create custom YUM repository
-        yum_repo = self._create(
-            user,
-            Repository,
+@pytest.fixture(scope='class')
```

Comment:
now that you've removed the class, this can be a module-level fixture

Reply:
Updated.